shell/browser/electron_browser_main_parts.cc

```diff
@@ -486,6 +487,9 @@ int ElectronBrowserMainParts::PreMainMessageLoopRun() {
 
   fake_browser_process_->PreMainMessageLoopRun();
 
+  ui::SelectFileDialog::SetFactory(
```

Comment:
Would be good to guard this behind `IS_WIN` buildflag since the current implementation is windows specific and I don't see `SelectFileDialog` guarding against when using the factory.